File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/PromiseValue.java
package com.github.dedis.popstellar.model.network.method.message.data.consensus;

import com.github.dedis.popstellar.model.Immutable;
import com.google.gson.annotations.SerializedName;

import java.util.Objects;

@Immutable
public final class PromiseValue {

  @SerializedName("accepted_try")
  private final int acceptedTry;

  @SerializedName("accepted_value")
  private final boolean acceptedValue;

  @SerializedName("promised_try")
  private final int promisedTry;

  public PromiseValue(int acceptedTry, boolean acceptedValue, int promisedTry) {
    this.acceptedTry = acceptedTry;
    this.acceptedValue = acceptedValue;
    this.promisedTry = promisedTry;
  }

  public int getAcceptedTry() {
    return acceptedTry;
  }

  public boolean isAcceptedValue() {
    return acceptedValue;
  }

  public int getPromisedTry() {
    return promisedTry;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    PromiseValue that = (PromiseValue) o;

    return acceptedTry == that.acceptedTry
        && acceptedValue == that.acceptedValue
        && promisedTry == that.promisedTry;
  }

  @Override
  public int hashCode() {
    return Objects.hash(acceptedTry, acceptedValue, promisedTry);
  }

  @Override
  public String toString() {
    return String.format(
        "PromiseValue{accepted_try=%s, accepted_value=%b, promised_try=%s}",
        acceptedTry, acceptedValue, promisedTry);
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/PromiseValue.kt
package com.github.dedis.popstellar.model.network.method.message.data.consensus

import com.github.dedis.popstellar.model.Immutable
import com.google.gson.annotations.SerializedName
import java.util.Objects

@Immutable
class PromiseValue(
    @field:SerializedName("accepted_try") val acceptedTry: Int,
    @field:SerializedName("accepted_value") val isAcceptedValue: Boolean,
    @field:SerializedName("promised_try") val promisedTry: Int
) {

  override fun equals(other: Any?): Boolean {
    if (this === other) {
      return true
    }
    if (other == null || javaClass != other.javaClass) {
      return false
    }
    val that = other as PromiseValue
    return acceptedTry == that.acceptedTry &&
        isAcceptedValue == that.isAcceptedValue &&
        promisedTry == that.promisedTry
  }

  override fun hashCode(): Int {
    return Objects.hash(acceptedTry, isAcceptedValue, promisedTry)
  }

  override fun toString(): String {
    return "PromiseValue{accepted_try=$acceptedTry, accepted_value=$isAcceptedValue, promised_try=$promisedTry}"
  }
}
